docs/architecture/adr-056-proving-amnesia-attacks.md

```diff
@@ -23,10 +24,98 @@ The currently only known form of global evidence stems from [flip flopping](http
+
+The latter of the two options was decided chiefly because it is important for the tendermint incentivisation mechanism that such behavior towards a light client is not only detectable but punishable. The logic that will need to be in place will involve the bare minimum to enable manual intervention. This therefore requires the on-chain submission of the faulty header that the light client witnessed plus the storing of vote sets by validators.
+
+## Detailed Design
+
+The first part of this short term solution is the `PotentialAmnesiaEvidence` data structure. This has the following properties that differ from its prior implementation.
+
+- It bundles all the malicious validators together instead of splitting them into individual evidence
+- It can be submitted on the chain
+- It does, by itself, not indicate which validators in the set misbehaved and should be slashed. It is merely a prompt for manual investigation.
+
+Before going any further, it is also important to mention that `PotentialAmnesiaEvidence` should only be formed from a valid attack on the light client and measures should be put in place to ensure that no node can easily forge the evidence as a means of spamming the network. 
+
+The data structure for `PotentialAmensiaEvidence` is as follows:
+
+```golang
+type PotentialAmnesiaEvidence struct {
+	*SignedHeader
+}
+```
+
+It conforms to the `Evidence` interface where `Time()` is the time of the heaer (note that this can be forged but must be within trusting period) and `Address()` returns the bytes of all validators in the commit
+
+`ValidateBasic()` inherit from the validate basic of the signed header
+
+### Verification of PotentialAmensiaEvidence
+
+`PotentialAmensiaEvidence` will be saved in the evidence pool and committed on chain if the following conditions are met.
+
+- The `SignedHeader` is valid -> `ValidateBasic()`
+
+- The `ValidatorsHash` of the header must be the same as the header that the node has committed. (else this is a lunatic attack)
+
+- The signatures of the commit must all be valid (for that header) `VerifyCommit()` using the validator set the node has for that height.
+
+- The header hash must be different to the hash of the header that the node has.
+
+- The `Commit` of the `SignedHeader` must be for a different round than the `Commit` of the header that the node has committed
+
+- The evidence must not have expired.
+
+
+The second part of the short term solution is saving `VoteSet`'s to the evidence pool. In order to avoid overflowing the validators memory, only the relevant information will be taken from the voteSets and formed into
+this specific struct
+
+```golang
+type VotesRecord struct {
+	votes []*Vote
```

Comment:
should we create a short version of the `Vote` type given that the height & round will be the same?

Reply:
also the type is known

Reply:
Perhaps we could save something similar to a `Commit`